can/listener.py

```diff
@@ -133,42 +124,41 @@ def stop(self):
         self.is_stopped = True
 
 
-if asyncio is not None:
-    class AsyncBufferedReader(Listener):
-        """A message buffer for use with :mod:`asyncio`.
+class AsyncBufferedReader(Listener):
+    """A message buffer for use with :mod:`asyncio`.
 
-        See :ref:`asyncio` for how to use with :class:`can.Notifier`.
-        
-        Can also be used as an asynchronous iterator::
+    See :ref:`asyncio` for how to use with :class:`can.Notifier`.
+    
+    Can also be used as an asynchronous iterator::
+
+        async for msg in reader:
+            print(msg)
+    """
+
+    def __init__(self, loop=None):
+        # set to "infinite" size
+        self.buffer = asyncio.Queue(loop=loop)
 
-            async for msg in reader:
-                print(msg)
+    def on_message_received(self, msg):
+        """Append a message to the buffer.
+        
+        Must only be called inside an event loop!
         """
+        self.buffer.put_nowait(msg)
 
-        def __init__(self, loop=None):
-            # set to "infinite" size
-            self.buffer = asyncio.Queue(loop=loop)
-
-        def on_message_received(self, msg):
-            """Append a message to the buffer.
-            
-            Must only be called inside an event loop!
-            """
-            self.buffer.put_nowait(msg)
-
-        def get_message(self):
-            """
-            Retrieve the latest message when awaited for::
-            
-                msg = await reader.get_message()
-
-            :rtype: can.Message
-            :return: The CAN message.
-            """
-            return self.buffer.get()
-
-        def __aiter__(self):
-            return self
+    def get_message(self):
+        """
+        Retrieve the latest message when awaited for::
         
-        def __anext__(self):
-            return self.buffer.get()
+            msg = await reader.get_message()
+
+        :rtype: can.Message
+        :return: The CAN message.
+        """
+        return self.buffer.get()
```

Comment:
Now we can mark this function `async` to make it clearer that it is an asynchronous function.

```python
async def get_message(self):
    return await self.buffer.get()
```